core/iwasm/common/wasm_runtime_common.c

```diff
@@ -7927,24 +7982,46 @@ wasm_runtime_inherit_imports(WASMModuleCommon *module,
     return 0;
 }
 
-void
-wasm_runtime_disinherit_imports(WASMModuleCommon *module,
-                                WASMExternInstance *imports, int32 import_count)
+WASMModuleInstanceCommon *
+wasm_runtime_instantiate_with_inheritance(
+    WASMModuleCommon *module, WASMModuleInstanceCommon *parent_inst,
+    WASMExecEnv *exec_env, uint32 stack_size, uint32 heap_size,
+    uint32 max_memory_pages, char *error_buf, uint32 error_buf_size)
 {
-#if WASM_ENABLE_INTERP != 0
-    if (module->module_type == Wasm_Module_Bytecode) {
-        wasm_disinherit_imports((WASMModule *)module, imports, import_count);
-        return;
+    int32 spawned_import_count = wasm_runtime_get_import_count(module);
+    if (spawned_import_count < 0) {
+        return NULL;
     }
-#endif
-#if WASM_ENABLE_AOT != 0
-    if (module->module_type == Wasm_Module_AoT) {
-        return aot_disinherit_imports((AOTModule *)module, imports,
-                                      import_count);
+
+    WASMExternInstance *spawned_imports = runtime_malloc(
+        sizeof(WASMExternInstance) * (uint64)spawned_import_count, NULL,
+        error_buf, error_buf_size);
```

Comment:
An improvement is to malloc memory only when spawned_import_count > 0, so as to avoid warning of allocating memory with size 0.

Reply:
Ok. I will finish ⬆️  in https://github.com/bytecodealliance/wasm-micro-runtime/pull/3898